pkg/providers/vsphere/reconciler/reconciler_test.go

```diff
@@ -251,6 +252,26 @@ func TestReconcilerReconcileControlPlaneSuccess(t *testing.T) {
 			},
 		},
 	)
+	tt.ShouldEventuallyExist(tt.ctx,
+		&controlplanev1.KubeadmControlPlane{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "workload-cluster",
+				Namespace: "eksa-system",
+			},
+		},
+	)
+	tt.ShouldEventuallyExist(tt.ctx,
+		&vspherev1.VSphereMachineTemplate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "workload-cluster-control-plane-1",
+				Namespace: "eksa-system",
+			},
+		},
+	)
+	capiCluster := capiCluster(func(c *clusterv1.Cluster) {
+		c.Name = "workload-cluster"
+	})
+	tt.ShouldEventuallyExist(tt.ctx, capiCluster)
```

Comment:
maybe check the secrets and config maps too? You can get the names by reading the ClusterResourceSet :)